isodatetime/tests/test_main.py

```diff
@@ -0,0 +1,245 @@
+    """Test isodatetime.main.main."""
+
+    @staticmethod
+    @patch('builtins.print')
+    def test_1_null(mock_print):
+        """Test calling usage 1, no or now argument."""
+        argv = sys.argv
+        with patch.object(
+            isodatetime.main.DateTimeOperator,
+            'process_time_point_str',
+            return_value='20190101T1234Z'
+        ):
+            for args in [[''], ['now']]:
+                sys.argv = args
+                try:
+                    isodatetime.main.main()
+                    mock_print.assert_called_with('20190101T1234Z')
+                finally:
+                    sys.argv = argv
+
+    @staticmethod
+    @patch('builtins.print')
+    def test_1_good(mock_print):
+        """Test calling usage 1, sample good arguments."""
+        env_ref = isodatetime.main.DateTimeOperator.ENV_REF
+        ref = os.environ.get(env_ref)
+        argv = sys.argv
+        for args, out in [
+            (['20200101T00Z'], '20200101T00Z'),
+            (['ref'], '20201225T0000Z'),
+            # UTC mode
+            (['-u', '20200101T00+0100'], '20191231T23+0000'),
+            (['--utc', '20200101T00+0100'], '20191231T23+0000'),
+            # With offsets
+            (['-s', 'P1D', '20191231T00Z'], '20200101T00Z'),
+            (['-s', 'P1D', '--offset=PT1H', '20191231T00Z'], '20200101T01Z'),
+            # Print format
+            (['-f', 'CCYY', '20191231T00Z'], '2019'),
+            (['--format', 'CCYY', '20191231T00Z'], '2019'),
+            (['--print-format', 'CCYY', '20191231T00Z'], '2019'),
+            (['20200101T00-1130', '-f', 'CCYYMMDDThhmm+0000'],
+             '20200101T1130+0000'),
+            # Parse format
+            (['--parse-format=%d/%m/%Y', '-f', 'CCYY-MM-DD', '31/12/2019'],
+             '2019-12-31'),
+            (['-p', '%d/%m/%Y', '-f', 'CCYY-MM-DD', '31/12/2019'],
+             '2019-12-31'),
+        ]:
+            sys.argv = [''] + args
+            os.environ[env_ref] = '20201225T0000Z'
+            try:
+                isodatetime.main.main()
+                mock_print.assert_called_with(out)
+            finally:
+                sys.argv = argv
+                if ref is not None:
+                    os.environ[env_ref] = ref
+                else:
+                    del os.environ[env_ref]
+
+    @patch('builtins.print')
+    def test_1_bad(self, mock_print):
+        """Test calling usage 1, sample bad arguments."""
+        argv = sys.argv
+        for args, out in [
+            # Bad time point string
+            (['201abc'], 'Invalid ISO 8601 date representation: 201abc'),
+            # Bad offset string
+            (['-s', 'add-a-year', '2019'], 'add-a-year: bad offset value'),
+        ]:
+            mock_print.reset_mock()
+            sys.argv = [''] + args
+            try:
+                with self.assertRaises(SystemExit) as ctxmgr:
+                    isodatetime.main.main()
+                mock_print.assert_not_called()
```

Comment:
Today-I-Learned again. Only used the internal properties of the mocked object, combined with `self.assertTrue` or `self.assertEqual`. Didn't know it had assert methods too. Thanks!!!